erpnext/support/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -49,26 +49,28 @@ def on_submit(self):
 
 			if d.sales_person not in email_map:
 				sp = frappe.get_doc("Sales Person", d.sales_person)
-				email_map[d.sales_person] = sp.get_email_id()
+				try:
+					email_map[d.sales_person] = sp.get_email_id()
+				except frappe.ValidationError:
+					pass
```

Comment:
Maybe it should send a msgprint saying the email ID is not valid and then pass. If it fails silently then the user might expect the event to be made.